### PR TITLE
optimize:add SQL exception of not supported by AT mode

### LINKED ISSUE #4324: AT mode handles unsupported SQL type exceptions
## Why you need it?
 Is your feature request related to a problem? Please describe in details
执行暂不支持的SQL类型时抛出异常
Exception should be thrown when executing an unsupported SQL
## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

### PR DESCRIPTION
**Pull request by GoodBoyCoder**
- [x] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
添加AT模式不支持的SQL语句异常提示

### Ⅱ. Does this pull request fix one issue?

fixes #4324

### CODE REVIEW

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/postgresql/PostgresqlOperateRecognizerHolder.java [hunk 1]
```diff
@@ -33,24 +33,39 @@
...
+            return postgresqlDeleteRecognizer;
+        }
+        return null;
     }
 
     @Override
     public SQLRecognizer getInsertRecognizer(String sql, SQLStatement ast) {
-        return new PostgresqlInsertRecognizer(sql, ast);
+        PostgresqlInsertRecognizer postgresqlInsertRecognizer = new PostgresqlInsertRecognizer(sql, ast);
+        if (postgresqlInsertRecognizer.isSqlSyntaxSupports(ast)) {
```
Comment (a364176773) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/postgresql/PostgresqlOperateRecognizerHolder.java:46]: 这段为什么不抽出来复用，入参为BaseRecognizer

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/oracle/BaseOracleRecognizer.java [hunk 2]
```diff
@@ -104,4 +114,56 @@ protected String getOrderByCondition(SQLOrderBy sqlOrderBy, final ParametersHold
...
+                throw new NotSupportYetException("not support the sql syntax with MergeStatement:" + x
+                        + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/sql-restrictions.html");
+            }
+
+            @Override
+            public boolean visit(SQLInsertStatement x) {
+                if (null != x.getQuery()) {
+                    //just like: insert into t select * from t1
+                    throw new NotSupportYetException("not support the sql syntax with insert into ... select:" + x
+                            + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/sql-restrictions.html");
```
Comment (lightClouds917) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/oracle/BaseOracleRecognizer.java:161]: 这个要不统一一下，都给出样例sql，或者都不给，然后引导到文档。

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java [hunk 3]
```diff
@@ -98,4 +106,58 @@ public void executeOrderBy(SQLOrderBy sqlOrderBy,SQLASTVisitor visitor) {
...
+                throw new NotSupportYetException("not support the sql syntax with MergeStatement:" + x
+                        + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/sql-restrictions.html");
+            }
+
+            @Override
+            public boolean visit(SQLInsertStatement x) {
+                if (null != x.getQuery()) {
+                    //just like: insert into t select * from t1
+                    throw new NotSupportYetException("not support the sql syntax with insert into ... select:" + x
+                            + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/sql-restrictions.html");
```
Comment (lightClouds917) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java:155]: 这个要不统一一下，都给出样例sql，或者都不给，然后引导到文档。

**Review by lightClouds917 — CHANGES_REQUESTED** [review]

**Comment by lightClouds917** [comment]
https://seata.io/zh-cn/docs/user/sqlreference/sql-restrictions.html文档里，大佬要不麻烦也给个样例sql。😄

File: sqlparser/seata-sqlparser-druid/src/test/java/io/seata/sqlparser/druid/DruidSQLRecognizerFactoryTest.java [hunk 4]
```diff
@@ -35,5 +38,56 @@ public void testSqlRecognizerCreation() {
...
+        Assertions.assertThrows(NotSupportYetException.class, () -> recognizerFactory.create(sql6, JdbcConstants.MYSQL));
+        Assertions.assertThrows(NotSupportYetException.class, () -> recognizerFactory.create(sql6, JdbcConstants.ORACLE));
+        Assertions.assertThrows(NotSupportYetException.class, () -> recognizerFactory.create(sql6, JdbcConstants.POSTGRESQL));
+
+        String sql7 = "delete t where id = ?";
+        Assertions.assertNotNull(recognizerFactory.create(sql7, JdbcConstants.MYSQL));
+        Assertions.assertNotNull(recognizerFactory.create(sql7, JdbcConstants.ORACLE));
+        Assertions.assertNotNull(recognizerFactory.create(sql7, JdbcConstants.POSTGRESQL));
+
+        String sql8 = "delete t where id in (select id from b)";
```
Comment (caohdgege) [inline sqlparser/seata-sqlparser-druid/src/test/java/io/seata/sqlparser/druid/DruidSQLRecognizerFactoryTest.java:78]: delete from

**Review by a364176773 — APPROVED** [review]
LGTM

**Comment by lightClouds917** [comment]
LGTM

**Review by lightClouds917 — APPROVED** [review]
LGTM

File: sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java [hunk 5]
```diff
@@ -98,4 +106,58 @@ public void executeOrderBy(SQLOrderBy sqlOrderBy,SQLASTVisitor visitor) {
...
+
+            @Override
+            public boolean visit(SQLInSubQueryExpr x) {
+                //just like: ...where id in (select id from t)
+                throw new NotSupportYetException("not support the sql syntax with InSubQuery:" + x
+                        + "\nplease see the doc about SQL restrictions https://seata.io/zh-cn/docs/user/sqlreference/dml.html");
+            }
+
+            @Override
+            public boolean visit(SQLSubqueryTableSource x) {
```
Comment (slievrly) [inline sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java:130]: for update ?

Comment (GoodBoyCoder) [reply sqlparser/seata-sqlparser-druid/src/main/java/io/seata/sqlparser/druid/BaseRecognizer.java:130]: That's right. And the example statement is to illustrate the scope of SQLSubqueryTableSource

**Pull request merged**